fix!: Support Assent v0.3.0 OIDC-based Google strategy

joshprice opened the pull request:
Users upgrading using Google auth strategy should be aware of these field name changes in the `user_info` map returned by Google:
  - `google_hd` → `hd` (hosted domain)
  - `email_verified` is now a `boolean` instead of a `string`

If user code accesses `user_info["google_hd"]`, it should be updated to `user_info["hd"]`."
  
Fixes #1097

joshprice committed:
fix: Support OIDC-based Google strategy introduced by assent 0.3.0

  - Now uses `Google.default_config([])` from `assent`, which provides:
    - `base_url: "https://accounts.google.com/"` (correct for OIDC discovery)
    - `authorization_params: [scope: "email profile"]`
    - `client_authentication_method: "client_secret_post"`

The old config set base_url: "https://www.googleapis.com", but since assent v0.3.0 the Google strategy uses OIDC which needs to fetch
configuration from https://accounts.google.com/.well-known/openid-configuration. The old base URL broke this discovery process.

Users upgrading should be aware of these field name changes in the user_info map returned by Google:
  - `google_hd` → `hd` (hosted domain)
  - `email_verified` is now a `boolean` instead of a `string`

  If user code accesses `user_info["google_hd"]`, it should be updated to `user_info["hd"]`."

maennchen reviewed (approved):
@joshprice I don't know the internals of the assent oidc strategy, but this implementation looks good to me.

joshprice committed:
chore: igniter upgrade script

zachdaniel commented:
This would be a breaking change then wouldn't it?

joshprice committed:
chore: formatting

joshprice commented:
@zachdaniel yes it's a breaking change, but also it's a fix to the assent upgrade that is broken in main (the release it was in was retired)

jimsynz committed:
chore: update assent to ~> 0.3.0 and regenerate DSL docs

- Remove version pin on assent (was pinned to < 0.3.0 for v4.x compatibility)
- Regenerate DSL documentation for Google and Auth0 strategies
